Src/main.c

```diff
@@ -836,54 +753,34 @@ void loadEEpromSettings()
         if (motor_kv < 300) {
             low_rpm_throttle_limit = 0;
         }
-        low_rpm_level = motor_kv / 100 / (32 / motor_poles);
+        low_rpm_level = motor_kv / 100 / (32 / eepromBuffer.motor_poles);
 
-        high_rpm_level = motor_kv / 12 / (32 / motor_poles);				
+        high_rpm_level = motor_kv / 12 / (32 / eepromBuffer.motor_poles);				
     }
     reverse_speed_threshold = map(motor_kv, 300, 3000, 1000, 500);
+
+    if (eepromBuffer.eeprom_version > 2) {
+        if (eepromBuffer.auto_advance > 1) {
+            eepromBuffer.auto_advance = 0;
+            save_flash_nolib(eepromBuffer.buffer, sizeof(eepromBuffer.buffer), eeprom_address);
```

Comment:
why re-save here? each save is a risk. If we lose power during a save the user can end up with incorrect settings.